Create view

sakamotomayu opened the pull request:
Create view(haml & scss)

sakamotomayu committed:
Install Haml & exchange from erg-file to haml-file

sakamotomayu committed:
Create view

Create view(haml & scss)

t-kusakabe commented on Gemfile:
```diff
@@ -45,3 +45,5 @@ group :development do
 end
 
+gem "font-awesome-rails"
+gem "haml-rails"
```
ファイルの最後は改行入れておきましょう！

sakamotomayu replied:
改行しました！

ryohma0510 replied:
gemのところもシングルクオーテーションに統一しましょう

sakamotomayu replied:
統一しました！

t-kusakabe commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,7 @@
+
+@import "./reset";
+@import "./messages";
+@import "font-awesome";
```
ここもです！

sakamotomayu replied:
ここも改行しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,157 @@
+#chat{
+  min-width: 960px;
+  .create_chat {
+  	background-color: #38AEF0;
```
インデントが深すぎます！

sakamotomayu replied:
修正しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,157 @@
+  	color: white;
+    p {
+  	text-align: center;
+  	}
```
ここずれてます！

sakamotomayu replied:
修正しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,157 @@
+  .chat {
+    min-width: 960px;
+    height: 100%;
+    .chat-side {
```
`&` が使えそう :eyes:

sakamotomayu replied:
&に書き換えました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,157 @@
+      }
+    }
+
+      .chat-group {
```
インデントずれてますよー！

t-kusakabe replied:
ここ修正できていますか？

sakamotomayu replied:
はい、修正したので問題ないかと思っていました＞＜

t-kusakabe commented on app/controllers/messages_controller.rb:
```diff
@@ -0,0 +1,6 @@
+class MessagesController < ApplicationController
+
+def index
```
インデント！

sakamotomayu replied:
修正しました！

t-kusakabe commented on app/views/layouts/application.html.haml:
```diff
@@ -0,0 +1,10 @@
+!!!
+%html
+  %head
+    %meta{:content => "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}/
```
ハッシュスペース、ハッシュロケットを意識しましょう！

sakamotomayu replied:
ハッシュロケットは使わない方がいいのでしょうか？
ハッシュスペースを入力しました！

sakamotomayu committed:
Create view(twice)

sakamotomayu committed:
Create view(third)

t-kusakabe commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,7 @@
+/*
+ *= require font-awesome
```
これいりますか？

sakamotomayu replied:
font-awesomeをインストールしてるからいらないんですね＞＜
削除しました！

t-kusakabe commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,7 @@
+/*
+ *= require font-awesome
+ */
+
```
reset cssが読み込まれてなiい気がします！

sakamotomayu replied:
「@import "./reset";」 と記述していたので、「@import "reset";」へ修正しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,169 @@
+    background-color: #38AEF0;
+    color: white;
+    p {
+  	  text-align: center;
```
インデントはスペース２つで！

sakamotomayu replied:
修正しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,169 @@
+    &-side {
+      height: 100vh;
+      width: 300px;
+      color: #fff;
```
16進記法は省略するのかしないのか、どっちかに統一しましょう！

sakamotomayu replied:
省略しない方に統一しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,169 @@
+      height: 100vh;
+      width: 300px;
+      color: #fff;
+      background-color: #2f3e51;
```
何色かパット見わかんないので変数化してあげましょ！

sakamotomayu replied:
変数化しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,169 @@
+      float: left;
+      width: calc(100% - 300px);
+      height: 100%;
+      .chat-header {
```
`&` 記法使いましょか!

sakamotomayu replied:
.chat-headerを&記法で書くとcssが当たらなくなるんですよね＞＜
.chat-headerの親要素が.chat-spaceだからでしょうか（子要素が.chat-space-headerではないからでしょうか）。

ryohma0510 replied:
そうですね。親要素の名前が&に収納されています

t-kusakabe replied:
```
.chat {
  &-space {}
  &-header {}
```
ってするのが無難ですかね！
@sakamotomayu

t-kusakabe replied:
@ryoma510 収納ではなく参照です。プログラミングにおいて言葉の表現は重要なので気をつけましょう :ok_woman:

sakamotomayu replied:
ありがとうございます！
.chat {
  &-space {}
  &-header {}
へ修正しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,169 @@
+            line-height: 20px;
+          }
+        }
+        &__right {
```
`&__right--edit-btn` ではだめですか？

t-kusakabe replied:
それと、btnは装飾ではない気が。。。
elementになりうると思うので `__` で繋げるのが適切かと！

sakamotomayu replied:
&__right--edit__btnに修正しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,169 @@
+        padding-right: 40px;
+        height: 520px;
+        background-color: #AAA;
+        .foo {
```
このclass名はだめです :no_good_woman:

sakamotomayu replied:
.chat__space__main__body__mainへ修正しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,169 @@
+        background-color: #AAA;
+        .foo {
+          padding-top: 46px;
+          span.chat-body__user {
```
spanいりますか？

sakamotomayu replied:
span削除しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,169 @@
+      }
+    }
+  }
+}
```
改行

sakamotomayu replied:
182行目を改行しました！

t-kusakabe commented on app/assets/stylesheets/user.scss:
```diff
@@ -0,0 +1,141 @@
+      float: left;
+      width: 50%;
+    }
+    &:after {
```
mixinしてあげましょ！

t-kusakabe commented on app/views/layouts/application.html.haml:
```diff
@@ -0,0 +1,11 @@
+!!!
+%html
+  %head
+    %meta{ :content => "text/html; charset=UTF-8", "http-equiv" => "Content-Type" }/
```
ハッシュロケットは撲滅しちゃいましょう！

t-kusakabe replied:
最後に `/` 置くのはXMLの習慣なのであまり適切ではないです！

sakamotomayu replied:
すべてシンボルで記述しました！
/も削除しました。

t-kusakabe commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,48 @@
+        .chat-myaccount__user
+          %span mayu
+        .chat-myaccount__action
+          -#link_to _path, :method => : do
```
なぜコメントアウト？

sakamotomayu replied:
すみません、コメントアウト外しました！

t-kusakabe commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,48 @@
+            %p Members:
+        .chat-header__right
+          .chat-header__right--edit-btn
+            %a{ :href => "リンク" }Edit
```
ヘルパー使いましょう！

sakamotomayu replied:
link_to使いました！

t-kusakabe commented:
セレクタがhtmlタグで指定されているのが気になります。。。
class名あててあげないとbem使う意味があまりなくなってしまうので工夫してみてください！

t-kusakabe commented:
あとスクショはこのissueに貼っていただけると！

sakamotomayu committed:
Create view(forth)

sakamotomayu committed:
Create view(fifth)

指摘いただいたところ修正

ryohma0510 commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+$neary-black: #434a54;
+
+
+#chat{
```
#chatと{の間にスペース入れましょう

sakamotomayu replied:
スペース入れました！

ryohma0510 commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+          }
+        }
+        &__right--edit__btn {
+            font-size: 14px;
```
インデントずれてないですか？

sakamotomayu replied:
修正しました！

ryohma0510 commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+          }
+        }
+        &__message {
+            font-size: 14px;
```
インデントずれてないですか？

sakamotomayu replied:
修正しました！

ryohma0510 commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+
+    }
+  }
+}
```
sublimetextの設定で保存時に勝手に最終行に改行を入れてくれるので入れてください！

t-kusakabe replied:
@sakamotomayu プラグインでぐぐるとでてきます :ok_man:

sakamotomayu replied:
入れました、ありがとうございます！

ryohma0510 commented on app/assets/stylesheets/user.scss:
```diff
@@ -0,0 +1,141 @@
+    padding: 0;
+    font-size: 13px;
+  }
+}
```
deviseのビューがコミットされてませんか？

sakamotomayu replied:
user.scssファイルを削除しました！

ryohma0510 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,48 @@
+          .chat-header__right--edit__btn
+            =link_to 'Edit', '_path', class: "edit_btn"
+      .chat-body
+        .foo
```
BEM無視してないですか？fooってわかりにくい気が・・・

sakamotomayu replied:
.chat-body__mainに変更しました！

koutaodawara commented on app/assets/stylesheets/user.scss:
```diff
@@ -0,0 +1,141 @@
+    border: 1px solid #ddd;
+    border-radius: 5px;
+
+    .account-page__inner--left, .account-page__inner--right {
```
&使いましょうか

sakamotomayu replied:
このファイルはdevise関連のため削除しました＞＜

ryohma0510 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,48 @@
+          -#link_to _path, :method => : do
+          %i.fa.fa-picture-o.fa-lg
+        .chat-footer__send
+          %input{ type: "submit", value: "Send", id: "aaa", name:  "send", class: "submit"}
```
基本シングルクオーテーションに統一しましょう

sakamotomayu replied:
統一しました！

ryohma0510 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,48 @@
+            %p Members:
+        .chat-header__right
+          .chat-header__right--edit__btn
+            =link_to 'Edit', '_path', class: "edit_btn"
```
基本シングルクオーテーションに統一しましょう

sakamotomayu replied:
シングルに統一しました！

ryohma0510 commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,4 @@
+@import "reset";
+@import "./messages";
+@import "font-awesome";
```
基本シングルクオーテーションに統一しましょう

sakamotomayu replied:
統一しました！

koutaodawara commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+        padding-right: 40px;
+        height: calc(100vh - 197px);
+        background-color: $gray1;
+        .foo {
```
クラス名

sakamotomayu replied:
変更しました！

koutaodawara commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+
+    }
+  }
+}
```
改行

sakamotomayu replied:
改行しました!

koutaodawara commented on app/views/layouts/application.html.erb:
```diff
@@ -5,6 +5,7 @@
   <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
   <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
   <%= csrf_meta_tags %>
+  <link href="app/assets/font-awesome.min.css" rel="stylesheet">
```
シングルクウォート

sakamotomayu replied:
シングルにしました

sakamotomayu committed:
Create view(sixth)

指摘いただいたところ修正

sakamotomayu committed:
Create view(seventh)

指摘いただいたところ修正

ryohma0510 commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+    position: absolute;
+    background-color: $light-blue;
+    color: white;
+  	text-align: center;
```
インデントずれてないですか？

sakamotomayu replied:
修正しました！

sakamotomayu committed:
Create view(eighth)

指摘いただいたところ修正

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+$light-blue: #38aef0;
```
変数化するものはそれだけのファイルを作って上げたほうがいいかもです！

sakamotomayu replied:
_color.scssを作ってそこに変数を入れました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+    text-align: center;
+  }
+
+  .chat {
```
インデントか括弧の位置がずれているかと！

sakamotomayu replied:
.chat-groupのインデントを修正したのでここは問題ないかと思うのですがいかがでしょうか＞＜

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+      height: 100%;
+      .chat-header {
+        height: 100px;
+        padding-left: 40px;
```
`padding: 0 40px;` だと1行減らせるのこっちの方がいいかと！
どれだけ行数少なく出来るかも大事です！(可読性が落ちない範囲で)

sakamotomayu replied:
なるほど！padding: 0 40px; に修正しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+        padding-right: 40px;
+        border-bottom: 1px solid $light-gray;
+        &__left {
+          &--group {
```
groupはmodifierですか？

sakamotomayu replied:
elementですね＞＜修正しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+            color: $deep-gray;
+            line-height: 50px;
+          }
+          &--member {
```
ここもelementかと！

sakamotomayu replied:
修正しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+          border: solid 2px $light-blue;
+          color: $light-blue;
+          text-decoration: none;
+          .edit_btn {
```
ここも工夫すると `&` 使えます！

sakamotomayu replied:
＆使いました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+        }
+      }
+
+      .chat-body {
```
親をIDじゃなくてclass名にしておくとここも `chat` 省略できますね！
これだとbemの恩恵が受けれていないです><

sakamotomayu replied:
すべてchat省略できるように書き換えました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+        position: relative;
+        padding-left: 40px;
+        padding-right: 40px;
+        height: calc(100vh - 197px);
```
197pxってなんの数値かわからないので説明変数使いましょう！

sakamotomayu replied:
headerとfooterを足した数が197pxなので、$header-footer-height: 197px;としました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+        background-color: $gray1;
+        &__main {
+          padding-top: 46px;
+          .chat-body__user {
```
この辺も `&` 使えます！

sakamotomayu replied:
＆使いました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+          margin: 10px 0 40px 0;
+        }
+      }
+
```
空行開けるタイミングが不規則なのがやや気になります。。。

sakamotomayu replied:
一応それぞれの箱の区切りとして空行を入れていたのですが、空行なしとしました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+        &__message {
+          font-size: 14px;
+          color: $neary-black;
+          margin: 10px 0 40px 0;
```
最後の0なくて良いです！

sakamotomayu replied:
そうですね、0削除しました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+          display: inline-block;
+          width: 80%;
+          min-width: calc(100% - 240px);
+          max-width: calc(100% - 240px);
```
minとmax指定する意味ってなんですかね？

sakamotomayu replied:
意味ないですね・・・
width: calc(100% - 240px);としました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+          width: 80%;
+          min-width: calc(100% - 240px);
+          max-width: calc(100% - 240px);
+          .textarea {
```
class名付けてあげましょう！

sakamotomayu replied:
クラス名をtextareaにしてみたのですが、わかりにくいでしょうか＞＜

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+          }
+        }
+        &__send {
+          .submit {
```
textareaと重複するところが多いのでまとめてあげると良さそうです！

sakamotomayu replied:
border-style: none;以外は他とまとめてみました！

t-kusakabe commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+          }
+        }
+      }
+
```
なんの空行でしょう？

sakamotomayu replied:
この空行も一応それぞれの箱の区切りとして入れていたのですが、空行なしとしました！

t-kusakabe commented on app/views/layouts/application.html.erb:
```diff
@@ -5,6 +5,7 @@
   <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
   <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
   <%= csrf_meta_tags %>
+  <link href='app/assets/font-awesome.min.css' rel='stylesheet'>
```
asssets pipelineで入ってませんかね？

t-kusakabe replied:
てかこのファイルのコミットがあるのがまずいですねｗ

sakamotomayu replied:
font-awesome.min.cssは検索しても出てきませんでしたので、削除しました。
この記述はデフォルトですよね？
どうしてコミットがあるとまずいのでしょうか？＞＜

t-kusakabe commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,48 @@
+        .chat-myaccount__user
+          %span mayu
+        .chat-myaccount__action
+          =link_to '/new' do
```
`=` の後スペースいれてあげましょう！

t-kusakabe replied:
pathがわかっているのならprefixで指定してあげましょう！

sakamotomayu replied:
スペース入れました！
まだrootパスしか指定していないので追って入力します＞＜

t-kusakabe commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,48 @@
+          %span テスト
+      .chat-footer
+        .chat-footer__message
+          %textarea{ type: 'textarea', name: 'text', id: 'aaa', placeholder: 'type a message', class: 'textarea' }
```
helper使えるといいですね^^

sakamotomayu replied:
~ text_area　と書いてみたのですが、エラーが出ます。
どこが間違っているでしょうか＞＜

sakamotomayu commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,182 @@
+    text-align: center;
+  }
+
+  .chat {
```
修正しました

sakamotomayu committed:
Create view(9th)

修正いただいたところ訂正、エラーの質問もあり

t-kusakabe commented:
すいません遅くなりました><
よいかと！
次に進んでください！！！

sakamotomayu committed:
Create views(10th)

text areaのヘルパーメソッドを用いて窮してみたがエラーが出たため(~ text_area type: 'textarea',
name: 'text', id: 'aaa', placeholder: 'type a message', class:
'textarea')、取り急ぎ通常の記述に戻した。

sakamotomayu committed:
Create views(11th)

text areaのヘルパーメソッドを用いて窮してみたがエラーが出たため(~ text_area type: 'textarea',
name: 'text', id: 'aaa', placeholder: 'type a message', class:
'textarea')、取り急ぎ通常の記述に戻した。
